src/Deprecation.php

```diff
@@ -93,7 +93,7 @@ public static function trigger(string $package, string $link, string $message, .
             self::$triggeredDeprecations[$link] = 1;
         }
 
-        if (self::$deduplication === true && self::$triggeredDeprecations[$link] > 1) {
+        if ((self::$deduplication ?? self::getDeduplicationFromEnv()) === true && self::$triggeredDeprecations[$link] > 1) {
```

Comment:
`self::$deduplication ?? self::getDeduplicationFromEnv()` is a repeating pattern. I'd make a `static` function for it that returns `self::$deduplication ?? ...whateverDerivedFromEnv...`